rs/replicated_state/src/canister_state/queues.rs

```diff
@@ -900,8 +880,10 @@ impl CanisterQueues {
         // persist it explicitly).
         if self.canister_queues.is_empty() && self.ingress_queue.is_empty() {
```

Comment:
Do we have a test that executes this garbage collection path or garbage collection more generally?

Reply:
We do. We have a bunch of tests around input schedules and GC-ed input queues. And a quite comprehensive test of garbage collection itself (queues with messages, queues with slot reservations, queues with none).